### PR TITLE
add subscription id telemetry

### PR DESCRIPTION
**Pull request by arjgupta**

### CODE REVIEW

File: src/Agent.Listener/Configuration/ConfigurationProvider.cs [hunk 1]
```diff
@@ -308,6 +314,30 @@ private async Task<List<DeploymentMachine>> GetDeploymentTargetsAsync(AgentSetti
 
             return machines;
         }
+
+        private string GetAzureSubscriptionId()
+        {
+            string azureSubscriptionId = string.Empty;
+            const string imdsUri = "http://169.254.169.254/metadata/instance/compute/subscriptionId?api-version=2017-08-01&format=text";
```
Comment (TingluoHuang) [inline src/Agent.Listener/Configuration/ConfigurationProvider.cs]: what is `http://169.254.169.254`? seem odd we hard code a http IP in  source code.

File: src/Agent.Listener/Configuration/ConfigurationProvider.cs [hunk 2]
```diff
@@ -308,6 +314,30 @@ private async Task<List<DeploymentMachine>> GetDeploymentTargetsAsync(AgentSetti
...
+                try
+                {
+                    azureSubscriptionId = httpClient.GetStringAsync(imdsUri).Result;
+                    if (!Guid.TryParse(azureSubscriptionId, out Guid result))
+                    {
+                        azureSubscriptionId = string.Empty;
+                    }
+                }
+                catch (Exception)
+                {
```
Comment (TingluoHuang) [inline src/Agent.Listener/Configuration/ConfigurationProvider.cs]: do we want to trace exception?

File: src/Agent.Listener/Configuration/ConfigurationProvider.cs (hunk 1 above)
Comment (arjgupta) [reply src/Agent.Listener/Configuration/ConfigurationProvider.cs]: Azure Instance Metadata Service is a REST endpoint available to all Azure VM's. This endpoint is a non-routable well known address ( 169.254.169.254 ). We are using this in order to fetch metadata about the subscription used to provision that VM. [More here](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service)

File: src/Agent.Listener/Configuration/ConfigurationProvider.cs (hunk 2 above)
Comment (arjgupta) [reply src/Agent.Listener/Configuration/ConfigurationProvider.cs]: done

**Comment by TingluoHuang** [comment]
@arjgupta do we need to worries about the subscription id is PII/EUII data?

File: src/Agent.Listener/Configuration/ConfigurationProvider.cs [hunk 3]
```diff
@@ -308,6 +314,34 @@ private async Task<List<DeploymentMachine>> GetDeploymentTargetsAsync(AgentSetti
 
             return machines;
         }
+
+        private string GetAzureSubscriptionId()
```
Comment (TingluoHuang) [inline src/Agent.Listener/Configuration/ConfigurationProvider.cs]: the caller `AddAgentAsync` is an async task.
change this to be an async task as well.
```C#
private async Task<string> GetAzureSubscriptionIdAsync()
{
     azureSubscriptionId  = await httpClient.GetStringAsync(imdsUri);
}
```

**Comment by arjgupta** [comment]
> @arjgupta do we need to worries about the subscription id is PII/EUII data?

@TingluoHuang no this is not PII/EUII

File: src/Agent.Listener/Configuration/ConfigurationProvider.cs (hunk 3 above)
Comment (arjgupta) [reply src/Agent.Listener/Configuration/ConfigurationProvider.cs]: done

**Pull request merged**